plugins/jobtype/src/azkaban/jobtype/HadoopSparkJob.java

```diff
@@ -348,9 +414,51 @@ private static void handleStandardArgument(Props jobProps,
       }
     }
 
+    info("Final classpath: " + classPath);
     return classPath;
   }
 
+  private String[] getSparkHomeConf() {
+    String sparkHome = null;
+    String sparkConf = null;
+    // If user has specified version in job property. e.g. spark.version=1.6.0
+    String jobSparkVer = getJobProps().get(SparkJobArg.SPARK_VERSION.azPropName);
+    if (jobSparkVer != null) {
+      // Spark jobtype supports this version through plugin's jobtype config
+      // e.g. spark.1.6.0.home=/path_to_spark/ in commonprivate.properties
+      sparkHome = getSysProps().get("spark." + jobSparkVer + ".home");
+      if (sparkHome != null) {
+        sparkConf = getSysProps().get("spark." + jobSparkVer + ".conf");
+        if (sparkConf == null) {
+          sparkConf = sparkHome + "/conf";
+        }
+        info("Using job specific spark: " + sparkHome + " and conf: " + sparkConf);
+        // Override the SPARK_HOME SPARK_CONF_DIR env for HadoopSecureSparkWrapper process(spark client)
+        getJobProps().put("env." + SPARK_HOME_ENV_VAR, sparkHome);
```

Comment:
Even though we configure spark home dir and spark conf dir in plugin properties, it would be better to add a check here for the existence of these directories.